### ISSUE TITLE
likwid-setFrequencies weird/unclear behavior with Uncore clock

### ISSUE THREAD

**Issue by rrzeschorscherl** [RRZE-HPC/likwid#145]
Version 4.3.1.

The help message says:
"The min/max Uncore frequency can be set freely, even to 0 or 1E20 but the hardware
stays inside its limits."

This is not true. tested on Haswell-EP E5-2695 v3:
```
$ likwid-setFrequencies --umin 0 --umax 3.0
ERROR - [./src/frequency.c:freq_setUncoreFreqMin:747] Success.
Given frequency 0 MHz lower than system limit of 1200 MHz
Setting of minimal Uncore frequency 0.000000 failed on socket 0

```
If I set them to the limits reported above,
```
$ likwid-setFrequencies --umin 1.2 --umax 2.301
```
there is no error message but the reported limits are now:
```
Current Uncore frequencies:
Socket 0: min/max 1.2/2.3 GHz
Socket 1: min/max 1.2/2.3 GHz
```
This is not what was expected - I expected UFS. Also I cannot measure an Uncore frequency beyond 2.3 GHz at this setting. After a reboot of the machine, the reported Uncore frequency range is:
```
Current Uncore frequencies:
Socket 0: min/max 1.2/3.0 GHz
Socket 1: min/max 1.2/3.0 GHz
```
Indeed, just after the reboot I can measure Uncore frequencies of close to 3 GHz. So, what is the official way to go back to UFS after setting the Uncore clock?

Cheers,
Georg.

**Comment by TomTheBear** [comment]
The help text mixed it up. Generally you can write whatever value you want in the register but the hardware ignores too high/low settings. LIKWID itself has its own limits to avoid malformed user input. I updated the help text.

LIKWID enforces some minimal and maximal possible uncore frequencies:
Minimum: Minimal CPU frequency (`likwid-setFrequencies -l`)
Maximum: Turbo frequency when running on a single core (C0 value at `likwid-powermeter -i`)
The previous maximum was the maximum CPU frequency which was not sufficient (like in your case). 

It still might be that the new maximum is not as high as the maximal possible setting (like directly after boot with UFS "enabled"). On a Haswell-EP E5-2695 v3 the maximal possible setting allowed by LIKWID would be 3.3 although the uncore's maximum is just 3.0.

Generally UFS is always enabled and in use as soon as the minimal and maximal Uncore frequency differ. If you "turn off" UFS in the BIOS it just sets minimal and maximal to the maximally possible frequency.

**Comment by TomTheBear** [comment]
This is fixed in 4.3.2
